index.js

```diff
@@ -39,18 +40,37 @@ async function getTds () {
  * @param {string} platform - platform to write
  * @param {object} config - the object to write
  */
-function writeConfigToDisk (platform, config, v1Config) {
+function writeConfigToDisk (platform, config) {
     let configName = platform
     if (platform.includes(BROWSERS_SUBDIR)) {
         configName = platform.replace(BROWSERS_SUBDIR, 'extension-')
     }
 
-    // Now that the features have been finalized, write out their hashes
+    // Convert config to backwards compatible versions
+    let compatConfig = config
+    const configsToWrite = []
+    for (const version of ['v2', 'v1']) {
+        if (!compatibility.compatFunctions[version]) {
+            continue
+        }
+
+        compatConfig = compatibility.compatFunctions[version](compatConfig)
+        addHashToFeatures(compatConfig)
+        configsToWrite.push({
+            version,
+            config: compatConfig
+        })
+    }
+
+    // Write backwards compatible versions to disk
+    for (const { version, config } of configsToWrite) {
+        fs.writeFileSync(`${GENERATED_DIR}/${version}/${configName}-config.json`, JSON.stringify(config, null, 4))
+    }
+
     addHashToFeatures(config)
-    addHashToFeatures(v1Config)
 
-    fs.writeFileSync(`${GENERATED_DIR}/v2/${configName}-config.json`, JSON.stringify(config, null, 4))
-    fs.writeFileSync(`${GENERATED_DIR}/v1/${configName}-config.json`, JSON.stringify(v1Config, null, 4))
+    // Write current version to disk
+    fs.writeFileSync(`${GENERATED_DIR}/v3/${configName}-config.json`, JSON.stringify(config, null, 4))
```

Comment:
Should this come out of the compatiblity module? or have something keep track of the current version so we can construct the directories from it and iterate backwards from n?

Reply:
Specifically I was talking about the `v3`.

Thinking about this some more, it would be nice if it was all part of the same for loop, starting with the latest version and counting down then on successive loops calling the relevant compat methods and writing the file.